core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -245,6 +248,13 @@ private void tryStart(Profiler profiler) {
             applyConfiguration(createCommand);
 
             containerId = createCommand.exec().getId();
+
+            PortForwardingContainer.INSTANCE.getNetwork().map(ContainerNetwork::getNetworkID).ifPresent(networkId -> {
```

Comment:
WDYT about refactoring to private method for better readibility? Took me some time to get the high level view what's happening here (and I possibly even got it wrong):

```
[...]
String networkMode = createCommand.getNetworkMode();
connectToPortForwardingNetwork(networkMode);
[...]



private void connectToPortForwardingNetwork(String networkMode) {
        PortForwardingContainer.INSTANCE.getNetwork().map(ContainerNetwork::getNetworkID).ifPresent(networkId -> {
            if (!Arrays.asList(networkId, "none", "host").contains(networkMode)) {
                dockerClient.connectToNetworkCmd().withContainerId(containerId).withNetworkId(networkId).exec();
            }
        });
    }

```

Reply:
👍 yes, I think that would help the readability.

Reply:
@kiview extracted